### ISSUE TITLE
Mouse position detected incorrectly

### ISSUE THREAD

**Issue by unfa** [OpenMusicKontrollers/synthpod#17]
I tested this in Carla plugin host.

Seems to affect Moony, Sherlock and other plugins aswell.
Here's a screenshot:
![spectacle tj4989](https://user-images.githubusercontent.com/491704/29193634-78067fa6-7e26-11e7-8ce7-9bfef7ac5438.png)

The plugin's window seems to detect the mouse position a bit down and to the left than th cursor is drawn onscreen by the X.org server. Interacting with the interface is very difficult.

**Comment by ventosus** [comment]
Thanks for reporting.

Do you have the same issue in other hosts, e.g. ardour, qtractor, jalv, ingen, or is it just a carla thing?

**Comment by unfa** [comment]
I haven't tested yet. I'll try in Ardour soon and report :)

**Comment by ventosus** [comment]
Well, I cannot reproduce this with carla 1.9.7 (2.0-beta5) here

**Comment by ventosus** [comment]
And ... running synthpod as a plugin itself has been deprecated for the time being, focus is to make it a better host.

**Comment by unfa** [comment]
Ok, I see the same behaviour in Ardour too.

Since this is not a priority, I'll wait and see if it'll go away with some updates to Carla or maybe other libraries.

**Comment by ventosus** [comment]
carla and ardour have different ways of handling plugin UIs. ardour/qtractor/jalv/ingen use a common library [1] to ahndle plugin UIs. carla (like synthpod host) does its own thing.

As you have the problem in two apps which use different backends to handle plugin UIs, it is likely something on my side that may be going wrong here.

It looks like the plugin UI does not resize properly into the host's parent window. What happens if you resize the host window my plugin UI is embedded in in the corresponding host (carla/ardour)?

There are up-to-date binaries of my plugins at https://dl/open-music-kontrollers.ch. Worth a try, too.

**Comment by ventosus** [comment]
Do you have the same issue with synthpod's UI when run as host, too?

    synthpod_jack -g

What about if you inject e.g. moony/sherlock into synthpod and open its UI (Shift-Right click)?

**Comment by ventosus** [comment]
Have you by accident an environment varialbe NK_SCALE set to a value different from 1?

    echo $NK_SCALE

I've found a bug in initial resize/scaling if users should set this value.

**Comment by ventosus** [comment]
Or you have a screen DPI different from 96...

**Comment by ventosus** [comment]
fixed in https://github.com/OpenMusicKontrollers/synthpod/commit/b067d8ec2f789cde7eac3184385cab03b39b76d0 (and in all other projects)

Find up-to-date binaries (unstable directories) at [1]

[1] https://dl.open-music-kontrollers.ch

**Comment by unfa** [comment]
1. My screen DPI is 96
2. echo $NK_SCALE returns nothing

After a recent update to Carla the problems seems to go away at leas on one system I use. Will check on another one too.

When I resize the window in Carla - the mouse works fine, there's no gap now, but the content of the window doesn't fit the window size:

![selection_999 109](https://user-images.githubusercontent.com/491704/29716292-16ea8312-89ab-11e7-9dc8-56e7eb129fe9.png)

That's not a showstopper though.

I'm really excited about this project!

**Comment by unfa** [comment]
Ok, on another system (also running Linux Mint 18.2 with KDE5 and newest Carla) the problems persists. This must be dependant on something else.
Standalone Synthpod works fine and has no problems resizing the window content when I maximize etc.

**Comment by ventosus** [comment]
Still, this is quite strange, and shouldn't be. My plugin UI's depend on X11 resize events from the parent window to properly resize. Those are properly sent in synthpod standalone. Will need to read carla's code, whether it does something differently there.

**Comment by ventosus** [comment]
I've added some code to all my plugin UIs for hosts that do not seem to send proper X11 resize events, may help in this issue, too, once synthpod as plugin will be suppported, again

https://github.com/OpenMusicKontrollers/synthpod/commit/6a1e660c349ca82340eb4bc5b626d4996c11903d
